tests/integration/test_task.py

```
@@ -80,3 +83,229 @@ def _run(self, input_files):
+    )
+
+    return {task}
+
+
+def test_py_diagnostic_run_sequential_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask sequentially with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        _run_tasks_sequential(tasks)
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask parallel with bad Python script."""
+    diagnostic_text = "import os\n\nprint(cow)"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    with pytest.raises(DiagnosticError) as err_mssg:
+        for task in tasks:
+            task.run()
+    exp_mssg = "diag_cow.py failed with return code 1"
+    assert exp_mssg in str(err_mssg.value)
+
+
+def test_py_diagnostic_run_parallel_task(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK Python script."""
+    diagnostic_text = "import os\n\nprint('cow')"
+    tasks = _get_diagnostic_tasks(tmp_path, diagnostic_text, 'py')
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    monkeypatch.setattr(BaseTask, '_run', _run)
+
+    for task in tasks:
+        task.run()
+
+
+def test_ncl_diagnostic_run_parallel_task_fails(monkeypatch, tmp_path):
+    """Run DiagnosticTask in parallel with OK NCL script."""
+    diagnostic_text = "cows on the [river]"
+
+    def _run(self, input_filesi=[]):
+        print(f'running task {self.name}')
+
+    if shutil.which('ncl') is not None:
```

It would be better to skip the test if ncl is not installed, rather than silently passing it. You can use [`pytest.mark.skipif`](https://docs.pytest.org/en/latest/skipping.html#id1) to do that.